src/pose-sensor.js

```
@@ -0,0 +1,170 @@
+      } else if (error.name === 'ReferenceError') {
+        // Fallback to devicemotion
+        this.api = 'devicemotion';
+        this.useDeviceMotion();
+      }
+    }
+
+    if (sensor) {
+      this.sensor = sensor;
+      this.sensor.addEventListener('reading', this._onSensorRead);
+      this.sensor.start();
+      this.api = 'sensor';
+    }
+
+    window.addEventListener('orientationchange', this._onOrientationChange);
+  }
+
+  useDeviceMotion() {
+    this.api = 'devicemotion';
+    this.fusionSensor = new FusionPoseSensor(this.config.K_FILTER,
+                                             this.config.PREDICTION_TIME_S,
+                                             this.config.YAW_ONLY,
+                                             this.config.DEBUG);
+  }
+
+  getOrientation() {
+    if (this.fusionSensor) {
+      return this.fusionSensor.getOrientation();
+    }
+
+    if (!this.sensor.quaternion) {
+      this._out[0] = this._out[1] = this._out[2] = 0;
+      this._out[3] = 1;
+      return this._out;
+    }
+
+    // Convert to THREE coordinate system: -Z forward, Y up, X right.
+    const q = this.sensor.quaternion;
+    this._sensorQ.set(q[0], q[1], q[2], q[3]);
+
+    const out = this._outQ;
+    out.copy(SENSOR_TO_VR);
+    out.multiply(this._sensorQ);
+    out.multiply(this._worldToScreenQ);
+
+    // Handle the yaw-only case.
+    if (this.config.YAW_ONLY) {
+      // Make a quaternion that only turns around the Y-axis.
+      out.x = 0;
+      out.z = 0;
+      out.normalize();
+    }
+
+    this._out[0] = out.x;
+    this._out[1] = out.y;
+    this._out[2] = out.z;
+    this._out[3] = out.w;
+    return this._out;
+  }
+
+  _onSensorError(e) {
+    this.error = e.error;
+    if (e.error.name === 'NotAllowedError') {
+      console.error('Permission to access sensor was denied');
+    } else if (e.error.name === 'NotReadableError' ) {
+      console.error('Sensor not present');
+    } else {
+      console.error(e.error);
+    }
```

IMO, we should return these errors